pom.xml

```diff
@@ -306,6 +311,14 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-oauth2-resource-server</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.security.oauth</groupId>
+            <artifactId>spring-security-oauth2</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.security.oauth.boot</groupId>
+            <artifactId>spring-security-oauth2-autoconfigure</artifactId>
```

Comment:
This is deprecated in Spring Boot 2.1 and is no longer recommended.

Reply:
Yes, I know. I tried to replace `UserInfoTokenServices` with `RemoteTokenServices` but this has proven to be difficult with Keycloak. So the plan is: fix OAuth2 first, and than get rid of spring-`security-oauth2-autoconfigure` as deprecated is better than broken.